credentials/alts/internal/handshaker/handshaker.go

```diff
@@ -156,8 +154,8 @@ func NewServerHandshaker(ctx context.Context, conn *grpc.ClientConn, c net.Conn,
 // ClientHandshake starts and completes a client ALTS handshake for GCP. Once
 // done, ClientHandshake returns a secure connection.
 func (h *altsHandshaker) ClientHandshake(ctx context.Context) (net.Conn, credentials.AuthInfo, error) {
-	if !clientHandshakes.TryAcquire(1) {
-		return nil, nil, errDropped
+	if err := clientHandshakes.Acquire(ctx, 1); err != nil {
+		return nil, nil, err
```

Comment:
Can you please clarify how returning the actual error help queuing instead of dropping?

Reply:
In the current code, we call `TryAcquire` on the semaphore. If the semaphore is capped out, then this immediately returns false, so we immediately return `errDropped` to the user.

When we replace the `TryAcquire` call with `Acquire` instead, we will block on trying to acquire the semaphore until `ctx` times out. Internally to the semaphore, there is a queue of "acquire attempts", so calling `Acquire` effectively adds the current goroutine to this queue.

As explained in b/312467484, we want to do this so that we have uniform behavior among the 3 languages.

Does that make sense / help clarify?

Reply:
Thanks for the clarification. This makes sense. Can you please add a comment that `Acquire` blocks until it can acquire? Thank you!